src/MidnightExample/HSLedger.agda

```diff
@@ -1,18 +1,16 @@
 {-# OPTIONS --overlapping-instances #-}
 module MidnightExample.HSLedger where
 
-open import Prelude hiding (_++_)
+open import Data.Integer        using (ℤ)
+open import Data.String         using (_++_)
+open import Interface.DecEq     using (DecEq)
+open import Interface.Hashable  using (Hashable; hash; Hashable₁; Hashable₂)
+open import Prelude             using (ℕ; String; List; _$_; foldr; _×_; _,_; Maybe)
```

Comment:
I think that generally it's best to keep modifications on module imports (`using/hiding/renaming`) to a minimum. Maybe my policy is bad, but it is currently to only use them to avoid name clashes and to rename things to a name that makes more sense in the current context (like renaming `×` to `∧`). I view the former as a bad thing (maybe this needs a type class) and the latter as a good thing. But making all imported names explicit hides both of those under a bunch of noise, and I'm not sure the noise adds anything.

I'm happy to discuss this further if you disagree. We should also really make some notes on code style somewhere.

Reply:
I think these are valid points.  I got in the habit of making explicit everything I'm using in every module, mainly for clarity and safety.  I think it's safer to name everything that's used in a module because, for example, you are being explicit about which `refl` you're using. I could imagine a case in which you use some term in a proof that inhabits a type from a different module than you thought, and consequently the proof is at best misleading, possibly just plain wrong, even though it type-checks.

I'll continue going through the `Ledger` module and explicitly importing stuff if only because it's a good exercise for me as I learn what's involved in each module as I go... but I won't make a pull request for those edits, unless you want to see them.

Reply:
I'm not sure that helps with safety, since you'd have to be aware of the entire import list while writing the module to be really sure what each name means. `M-.` is much faster and requires less context to do the same thing.